packages/@lwc/engine-server/src/renderer.ts

```diff
@@ -56,9 +56,7 @@ class HTMLElement {
         if (!name) {
             throw new TypeError(`Invalid Construction`);
         }
-        const elm = createElement(name);
-        elm.constructor = this.constructor; // restoring constructor into the fake instance
```

Comment:
since we don't rely on the constructor to extract the previously set `upgrade` value from setter, we can remove this hack here for SSR.